src/rpc/protocol.cpp

```diff
@@ -166,16 +180,32 @@ int ReadHTTPStatus(std::basic_istream<char>& stream, int &proto)
     boost::split(vWords, str, boost::is_any_of(" "));
     if (vWords.size() < 2)
         return HTTP_INTERNAL_SERVER_ERROR;
-    proto = 0;
-    const char *ver = strstr(str.c_str(), "HTTP/1.");
-    if (ver != nullptr && !ParseInt(std::string(ver+7), &proto)) {
-        throw std::invalid_argument("Unable to parse proto.");
+
+    size_t length = str.length();
+
+    // If CR is found at the end of the string, strip it, because ParseInt will fail if it is present.
+    if (str.substr(length - 1) == "\r") {
+        str = str.substr(0, length - 1);
+        --length;
+    }
+
+    size_t start_pos = str.find("HTTP/1.");
+
+    if (start_pos != std::string::npos && str.length() - start_pos > 7) {
+        str = str.substr(start_pos + 7);
+
+        if (!ParseInt(str, &proto)) {
+            error("%s: Unable to parse protocol in HTTP string: %s", __func__, str);
+            return HTTP_INTERNAL_SERVER_ERROR;
+        }
     }
 
     int status = 0;
     if (!ParseInt(vWords[1], &status)) {
-        throw std::invalid_argument("Unable to parse status.");
+        error("%s: Unable to parse status: %s", __func__, vWords[1]);
+        return HTTP_INTERNAL_SERVER_ERROR;
```

Comment:
```suggestion
        return -1;
```
nit: Server sending an invalid response should not be equivalent to a proper internal server error. Will need a tweak in `rpc/client.cpp`.

Reply:
See my comments above.

Reply:
I think for right now we should log an error in the debug log and do nothing (i.e. equivalent to status=0), which is what the original atoi code did. This duplicates the old API, right or not. We can put corrections and improvements into another PR. This PR is to simply fix the problem introduced by change from atoi to ParseInt.